enterprise/server/remote_execution/containers/docker/docker.go

```diff
@@ -160,50 +173,36 @@ func (r *dockerCommandContainer) Run(ctx context.Context, command *repb.Command,
 		}()
 	}()
 
-	r.copyContainerLogs(ctx, cid, result)
-	if result.Error != nil {
+	var stdout, stderr bytes.Buffer
+	eg := &errgroup.Group{}
+	eg.Go(func() error {
+		_, errCopy := stdcopy.StdCopy(&stdout, &stderr, hijackedResp.Reader)
+		return wrapDockerErr(errCopy, "failed to copy docker container output")
+	})
+	eg.Go(func() error {
+		statusCh, errCh := r.client.ContainerWait(ctx, cid, dockercontainer.WaitConditionNotRunning)
+		select {
+		case err := <-errCh:
+			return wrapDockerErr(err, "container did not exit cleanly")
+		case s := <-statusCh:
+			containerTerminated = true
+			if s.Error != nil {
+				return wrapDockerErr(status.UnknownError(s.Error.Message), "failed to get container status")
+			}
+			result.ExitCode = int(s.StatusCode)
+			return nil
+		}
+	})
+	if err := eg.Wait(); err != nil {
+		result.Error = wrapDockerErr(err, "failed to run docker container")
```

Comment:
avoid double-wrapping the error:

```suggestion
		result.Error = err
```